### ISSUE TITLE
FXIOS-1960 ⁃ Two Home tabs created when using cmd-t shortcut

### ISSUE THREAD

**Issue by aegatlin** [mozilla-mobile/firefox-ios#8707]
## Steps to reproduce

1. Have firefox open and focused
2. Press cmd-t

### Expected behavior

One new Home tab is generated

### Actual behavior

Two new Home tabs are generated

### Device & build information
* Device:
   - iPad Pro 12.9-inch 4th gen
   - Software version 14.6 (has existed the whole time I’ve used firefox on ipad, so earlier versions presumably are similarly buggy).

* Build version:
    - Firefox Daylight 34.0 (4920)

### Notes
Attachments:

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1960)

**Comment by gutley** [comment]
Note to Devs. While I don't have a real hardware keyboard on a real iPad to verify my findings, it is possible to replicate this in the Simulator. At which point the Logic error comes to light:

Assuming that in real life the way the key presses happen is:

1. Command Key goes down
2. Shortly after the 'T' key goes down
3. Shortly after the 'T' key is released
4. Shortly after the Command key is released

At stage 1, and this might be a surprise to you, this code in BrowserViewController triggers:

```swift
     override func pressesBegan(_ presses: Set<UIPress>, with event: UIPressesEvent?) {
         super.pressesBegan(presses, with: event)
         if #available(iOS 13.4, *) {
             guard let key = presses.first?.key, (key.keyCode == .keyboardLeftGUI || key.keyCode == .keyboardRightGUI) else { return } //GUI buttons = CMD buttons on ipad/mac
             self.isCmdClickForNewTab = true
         }
    }

```

Then at Stage 2, this then fires from TabTrayController+KeyCommands:
```swift
    @objc func didOpenNewTabKeyCommand() {
        TelemetryWrapper.recordEvent(category: .action, method: .press, object: .keyCommand, extras: ["action": "new-tab"])
        openNewTab()
    }
```
and a new tab is created.

When the new tab is created, BrowserViewController.webview:decidePolicy... fires and hits this bit of code:

```swift
        if isCmdClickForNewTab {
            guard let url = webView.url, let isPrivate = self.tabManager.selectedTab?.isPrivate else { return }
            homePanelDidRequestToOpenInNewTab(url, isPrivate: isPrivate)
            self.isCmdClickForNewTab = false
            decisionHandler(.cancel)
        }
```

And because `isCmdClickForNewTab` was set in Stage 1 another tab is created.

Hence why we end up with 2 tabs instead of 1.

In the Simulator I have also seen cases where multiple tabs will be opened until you get past stage 4 above. But because Cmd-key handling in Simulator is a bit flakey I really don't know if this is a real world possibility.

Note that you will also get 2 tabs if you hold down the Command Key while pressing the "Create New Tab" '+' button in the top-right of the screen.

So to summarise - I don't think pressesBegin/End do what you expect. Looking at how all this comes together I think you think that those will only be called when the user physically touches the screen. But as we have seen, touching keys on the hardware keyboard also triggers them.

**Comment by SimonBasca** [comment]
This is a duplicate of  #7519
